src/btree/bt_debug.c

```diff
@@ -480,26 +476,12 @@ __debug_hs_key(WT_DBG *ds)
      * Open a history store cursor positioned at the end of the data store key (the newest record)
      * and iterate backwards until we reach a different key or btree.
      */
-    hs_cursor = session->hs_cursor;
-    hs_cursor->set_key(hs_cursor, hs_btree_id, ds->key, WT_TS_MAX, WT_TXN_MAX);
-    ret = hs_cursor->search_near(hs_cursor, &exact);
-
-    /* If we jumped to the next key, go back to the previous key. */
-    if (ret == 0 && exact > 0)
-        ret = hs_cursor->prev(hs_cursor);
-
-    for (; ret == 0; ret = hs_cursor->prev(hs_cursor)) {
-        WT_RET(hs_cursor->get_key(hs_cursor, &hs_btree_id, ds->t1, &older_start_ts, &hs_counter));
-
-        if (hs_btree_id != btree->id)
-            break;
-
-        WT_RET(__wt_compare(session, NULL, ds->key, ds->t1, &cmp));
-        if (cmp != 0)
-            break;
+    hs_cursor->set_key(hs_cursor, 4, hs_btree_id, ds->key, WT_TS_MAX, WT_TXN_MAX);
```

Comment:
We actually need the HS cursor here. I am curious why you are not opening a HS cursor here instead of passing it around as function parameter?

Reply:
Opening an hs key for each key is too much. At least we should batch it for each page.

Reply:
Makes sense.